### PR TITLE
[JENKINS-61661] Better error messages in case of an InvalidCacheLoadException when no Jira credentials are provided

### PR DESCRIPTION
**Pull request by nussera**

### CODE REVIEW

**Comment by nussera** [comment]
Some tests already fail on me when I try to build the master branch. Most likely this has something to do with my network settings. My changes didn't trigger any new tests to fail :)

File: src/main/java/hudson/plugins/jira/JiraSite.java [hunk 1]
```diff
@@ -872,7 +873,7 @@ public JiraIssue getIssue(final String id) throws IOException {
             Optional<Issue> issue = issueCache.get(id, () -> {
                 JiraSession session = getSession();
                 if (session == null) {
-                    return null;
+                    return Optional.absent();
```
Comment (warden) [inline src/main/java/hudson/plugins/jira/JiraSite.java:876]: Can you add a test for this change?

**Comment by warden** [comment]
The t[ests are currently failing](https://ci.jenkins.io/job/Plugins/job/jira-plugin/job/PR-229/2/display/redirect) and [the test currently doesn't cover the changed lines](https://coveralls.io/builds/30190398/source?filename=src%2Fmain%2Fjava%2Fhudson%2Fplugins%2Fjira%2FJiraSite.java#L876).

**Comment by nussera** [comment]
Why are the tests failing there? I can't see a relation between my changed code and the failed tests. I mean the 'travis' tests succeeded. Shouldn't they have failed too?

**Comment by warden** [comment]
Good point, it seems that was caused by some cloud hiccup.
Travis is currently not running the tests, it's only running the coverage..
```
[2020-04-21T12:19:01.392Z] [INFO] -------------------------------------------------------
[2020-04-21T12:19:01.392Z] [INFO]  T E S T S
[2020-04-21T12:19:01.392Z] [INFO] -------------------------------------------------------
[2020-04-21T12:19:02.207Z] [INFO] Running InjectedTest
[2020-04-21T12:19:18.832Z] [INFO] Tests run: 44, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 15.394 s - in InjectedTest
[2020-04-21T12:19:18.832Z] [INFO] Running com.atlassian.httpclient.apache.httpcomponents.ApacheAsyncHttpClientTest
[2020-04-21T12:19:20.500Z] EC2 (aws) - Ubuntu 18.04 LTS (i-04922351d3ce92cd2) was marked offline: Connection was broken: java.util.concurrent.TimeoutException: Ping started at 1587471229168 hasn't completed by 1587471469169
[2020-04-21T12:19:20.500Z] 	at hudson.remoting.PingThread.ping(PingThread.java:133)
[2020-04-21T12:19:20.500Z] 	at hudson.remoting.PingThread.run(PingThread.java:89)
[2020-04-21T12:19:20.500Z] 
[2020-04-21T12:19:20.501Z] [Pipeline] // node
[2020-04-21T12:19:20.534Z] [Pipeline] }
[2020-04-21T12:19:20.534Z] Failed in branch linux-11
[2020-04-21T12:19:20.589Z] Sending interrupt signal to process
[2020-04-21T12:19:21.356Z] [INFO] 
[2020-04-21T12:19:21.356Z] [INFO] Results:
[2020-04-21T12:19:21.356Z] [INFO] 
[2020-04-21T12:19:21.356Z] [INFO] Tests run: 44, Failures: 0, Errors: 0, Skipped: 0
[2020-04-21T12:19:21.356Z] [INFO] 
[2020-04-21T12:19:21.739Z] [ERROR] There are test failures.
[2020-04-21T12:19:21.739Z] 
[2020-04-21T12:19:21.739Z] Please refer to /home/jenkins/workspace/Plugins_jira-plugin_PR-229/target/surefire-reports for the individual test results.
[2020-04-21T12:19:21.739Z] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[2020-04-21T12:19:21.739Z] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[2020-04-21T12:19:21.739Z] Command was /bin/sh -c cd /home/jenkins/workspace/Plugins_jira-plugin_PR-229 && /home/jenkins/tools/hudson.model.JDK/jdk8/jdk8u242-b08/jre/bin/java -Xmx768M -Djava.awt.headless=true -jar /home/jenkins/workspace/Plugins_jira-plugin_PR-229/target/surefire/surefirebooter1762810771478227585.jar /home/jenkins/workspace/Plugins_jira-plugin_PR-229/target/surefire 2020-04-21T12-19-00_334-jvmRun1 surefire7253532333121414707tmp surefire_04793388657619781759tmp
[2020-04-21T12:19:21.739Z] Error occurred in starting fork, check output in log
[2020-04-21T12:19:21.739Z] Process Exit Code: 143
[2020-04-21T12:19:21.739Z] Crashed tests:
[2020-04-21T12:19:21.739Z] com.atlassian.httpclient.apache.httpcomponents.ApacheAsyncHttpClientTest
[2020-04-21T12:19:21.739Z] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[2020-04-21T12:19:21.739Z] Command was /bin/sh -c cd /home/jenkins/workspace/Plugins_jira-plugin_PR-229 && /home/jenkins/tools/hudson.model.JDK/jdk8/jdk8u242-b08/jre/bin/java -Xmx768M -Djava.awt.headless=true -jar /home/jenkins/workspace/Plugins_jira-plugin_PR-229/target/surefire/surefirebooter1762810771478227585.jar /home/jenkins/workspace/Plugins_jira-plugin_PR-229/target/surefire 2020-04-21T12-19-00_334-jvmRun1 surefire7253532333121414707tmp surefire_04793388657619781759tmp
[2020-04-21T12:19:21.739Z] Error occurred in starting fork, check output in log
[2020-04-21T12:19:21.739Z] Process Exit Code: 143
[2020-04-21T12:19:21.739Z] Crashed tests:
[2020-04-21T12:19:21.739Z] com.atlassian.httpclient.apache.httpcomponents.ApacheAsyncHttpClientTest
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:670)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:283)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1161)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1002)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:848)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[2020-04-21T12:19:21.739Z] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[2020-04-21T12:19:21.739Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2020-04-21T12:19:21.739Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2020-04-21T12:19:21.739Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2020-04-21T12:19:21.739Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2020-04-21T12:19:21.739Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[2020-04-21T12:19:21.739Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[2020-04-21T12:19:21.739Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[2020-04-21T12:19:21.739Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[2020-04-21T12:19:21.739Z] 
[2020-04-21T12:19:21.739Z] [INFO] 
[2020-04-21T12:19:21.739Z] [INFO] --- maven-license-plugin:1.8:process (default) @ jira ---
[2020-04-21T12:19:22.220Z] script returned exit code 143
[2020-04-21T12:19:22.222Z] [Pipeline] }
[2020-04-21T12:19:22.249Z] [Pipeline] // withEnv
[2020-04-21T12:19:22.276Z] [Pipeline] junit
[2020-04-21T12:19:22.309Z] Recording test results
[2020-04-21T12:19:22.650Z] [Pipeline] }
[2020-04-21T12:19:22.674Z] [Pipeline] // stage
[2020-04-21T12:19:22.696Z] [Pipeline] }
[2020-04-21T12:19:22.722Z] [Pipeline] // timeout
[2020-04-21T12:19:22.743Z] [Pipeline] isUnix
[2020-04-21T12:19:22.744Z] [Pipeline] sh
[2020-04-21T12:19:23.303Z] + docker system prune --force --all
[2020-04-21T12:19:23.686Z] Total reclaimed space: 0B
[2020-04-21T12:19:23.785Z] [Pipeline] }
[2020-04-21T12:19:23.806Z] [Pipeline] // node
[2020-04-21T12:19:23.826Z] [Pipeline] }
[2020-04-21T12:19:23.826Z] Failed in branch linux-8
[2020-04-21T12:19:23.843Z] [Pipeline] // parallel
[2020-04-21T12:19:23.864Z] [Pipeline] End of Pipeline
[2020-04-21T12:19:24.921Z] 
[2020-04-21T12:19:24.921Z] GitHub has been notified of this commit’s build result
[2020-04-21T12:19:24.921Z] 
[2020-04-21T12:19:24.921Z] Also:   hudson.model.Computer$TerminationRequest: Termination requested at Tue Apr 21 12:17:49 UTC 2020 by Thread[Ping thread for channel hudson.remoting.Channel@4bde8ff:EC2 (aws) - Ubuntu 18.04 LTS (i-04922351d3ce92cd2),5,main] [id=361536]
[2020-04-21T12:19:24.921Z] 		at hudson.model.Computer.recordTermination(Computer.java:227)
[2020-04-21T12:19:24.921Z] 		at hudson.model.Computer.disconnect(Computer.java:491)
[2020-04-21T12:19:24.921Z] 		at hudson.slaves.SlaveComputer.disconnect(SlaveComputer.java:753)
[2020-04-21T12:19:24.921Z] 		at hudson.slaves.ChannelPinger$1.onDead(ChannelPinger.java:198)
[2020-04-21T12:19:24.921Z] 		at hudson.remoting.PingThread.ping(PingThread.java:133)
[2020-04-21T12:19:24.921Z] 		at hudson.remoting.PingThread.run(PingThread.java:89)
[2020-04-21T12:19:24.921Z] Also:   org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsBodyExecution.cancel(CpsBodyExecution.java:253)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.steps.BodyExecution.cancel(BodyExecution.java:76)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.steps.ParallelStepExecution.stop(ParallelStepExecution.java:67)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.steps.ParallelStep$ResultHandler$Callback.checkAllDone(ParallelStep.java:150)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.steps.ParallelStep$ResultHandler$Callback.onFailure(ParallelStep.java:137)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsBodyExecution$FailureAdapter.receive(CpsBodyExecution.java:361)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.impl.ThrowBlock$1.receive(ThrowBlock.java:68)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.Next.step(Next.java:83)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
[2020-04-21T12:19:24.921Z] 		at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
[2020-04-21T12:19:24.921Z] 		at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
[2020-04-21T12:19:24.921Z] 		at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:185)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:400)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$400(CpsThreadGroup.java:96)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:312)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:276)
[2020-04-21T12:19:24.921Z] 		at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
[2020-04-21T12:19:24.921Z] 		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2020-04-21T12:19:24.921Z] 		at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
[2020-04-21T12:19:24.921Z] 		at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
[2020-04-21T12:19:24.921Z] 		at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
[2020-04-21T12:19:24.921Z] 		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2020-04-21T12:19:24.921Z] 		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2020-04-21T12:19:24.921Z] 		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2020-04-21T12:19:24.921Z] 		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2020-04-21T12:19:24.921Z] 		at java.base/java.lang.Thread.run(Thread.java:834)
[2020-04-21T12:19:24.921Z] java.lang.IllegalStateException: hudson.plugins.ec2.EC2OndemandSlave[EC2 (aws) - Ubuntu 18.04 LTS (i-04922351d3ce92cd2)] is offline
[2020-04-21T12:19:24.921Z] 	at org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution$PlaceholderTask$PlaceholderExecutable.run(ExecutorStepExecution.java:829)
[2020-04-21T12:19:24.921Z] 	at hudson.model.ResourceController.execute(ResourceController.java:97)
[2020-04-21T12:19:24.922Z] 	at hudson.model.Executor.run(Executor.java:428)
[2020-04-21T12:19:24.922Z] Finished: FAILURE

```
@nussera can you push any commit that will re-trigger the build again?

**Pull request merged**